### DISCUSSION TITLE
建议支持开源AI大模型API调用

### DISCUSSION THREAD

**Discussion by dynax4u** [immersive-translate/immersive-translate#1753] [Ideas]
### 新功能 | New Feature

1.支持LM STUDIO本地SERVER服务的API调用
2.支持openrouter API调用

### 为什么您想要添加这些功能 | Why you want to add these features

1.以上三种，都是比较流行的开源大模型本地/在线解决方案，对于翻译量大，专业翻译要求高的用户是刚需。因为开源大模型可以自行微调，加入相应的专业语料，以适应行业需求。
2.本地开源大模型提供的服务很稳定，不会受到各种外部网络服务状况的影响，运行有保障。
3.

### 提议 | Proposal

建议优先考虑实现并优化，对LM STUDIO本地SERVER服务的API调用，因为该软件对普通用户的使用门槛低，且支持CUDA/OPENCL/ROCm，能高效利用Nvidia和AMD显卡的算力。

**Comment by theowenyoung** [comment]
1.5.7 版本起，您可以在翻译服务里添加任意 openai 接口兼容的翻译服务：

![CleanShot 2024-05-29 at 22 32 09@2x](https://github.com/immersive-translate/immersive-translate/assets/62473795/90583b07-610b-45c3-90c1-d2c599267cbb)
